src/analysis/retail/shaman/restoration/modules/dragonflight/Tier31.tsx

```diff
@@ -0,0 +1,151 @@
+import { SHAMAN_T31_ID } from 'common/ITEMS/dragonflight';
+import { SpellLink } from 'interface';
+import ItemSetLink from 'interface/ItemSetLink';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import Statistic from 'parser/ui/Statistic';
+import { formatNumber, formatPercentage } from 'common/format';
+import ItemHealingDone from 'parser/ui/ItemHealingDone';
+import RiptideTracker from '../core/RiptideTracker';
+import Combatants from 'parser/shared/modules/Combatants';
+
+const RIPTIDE_4P_POWER = 0.2; // Riptide bonus power = 25% => Tier set value = 25%/125% = 20% of total amount
+//const debug = false;
+
+/**
+ * **Resto Shaman T31 (Amirdrassil)**
+ *
+ * 2pc: Chain Heal, Healing Surge, and Healing Wave mark their initial target with a Tidal Reservoir, causing them to receive 15% of all Riptide healing you deal for 15 sec.
+ *
+ * 4pc : Riptide's healing is increased by 25%. If Riptide is active on the same target as Tidal Reservoir, its heal over time effect has a 6% chance to create a new Riptide on a nearby ally.
+ */
+
+export default class Tier31 extends Analyzer {
+  static dependencies = {
+    riptideTracker: RiptideTracker,
+    combatants: Combatants,
+  };
+  protected riptideTracker!: RiptideTracker;
+  protected combatants!: Combatants;
+  has4pc: boolean;
+  tidalReservoirHealing: number = 0;
+  tidalReservoirOverhealing: number = 0;
+  bonus4pcRiptideHealing: number = 0;
+  percent4pcRiptideHealing: number = 0;
+  bonus4pcRiptideAmount: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.has2PieceByTier(TIERS.T31);
+    this.has4pc = this.selectedCombatant.has4PieceByTier(TIERS.T31);
+
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(talents.RIPTIDE_TALENT),
+      this.handleRiptideHeal,
+    );
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(ITEMS.T31_TIDAL_RESERVOIR_HEAL),
+      this.on2pcHeal,
+    );
+  }
+
+  get twoPieceOverhealPercent() {
+    return formatPercentage(
+      this.tidalReservoirOverhealing /
+        (this.tidalReservoirHealing + this.tidalReservoirOverhealing),
+    );
+  }
+
+  get fourPieceTotalHealing() {
+    return this.bonus4pcRiptideHealing + this.percent4pcRiptideHealing;
+  }
+
+  on2pcHeal(event: HealEvent) {
+    this.tidalReservoirHealing += event.amount + (event.absorbed || 0);
+    this.tidalReservoirOverhealing += event.overheal || 0;
+  }
+
+  handleRiptideHeal(event: HealEvent) {
+    if (this.has4pc) {
+      const targetId = event.targetID;
+      const spellId = event.ability.guid;
+
+      this.percent4pcRiptideHealing += (event.amount + (event.absorbed || 0)) * RIPTIDE_4P_POWER;
+
+      if (event.tick) {
+        if (!this.riptideTracker.hots[targetId] || !this.riptideTracker.hots[targetId][spellId]) {
+          return;
+        }
```

Comment:
i'm not sure what this is doing. if the condition is true, it immediately returns. if the condition is false, it exits the nested `if`s and then immediately returns. it feels like this is maybe some WIP code that was not totally cleaned up (or forgotten and not finished?)